war/src/main/java/org/exoplatform/addons/es/EmbeddedESStartupServlet.java

```diff
@@ -42,27 +46,37 @@ public void init() throws ServletException {
     }
 
     LOG.info("Initializing elasticsearch Node '" + getServletName() + "'");
-    Settings.Builder settings = Settings.settingsBuilder();
+    Settings.Builder settings = Settings.builder();
 
     InputStream resourceAsStream = getServletContext().getResourceAsStream("/WEB-INF/elasticsearch.yml");
-    if (resourceAsStream != null) {
+    if (resourceAsStream == null) {
+      throw new ServletException("Error while initializing elasticsearch node, configuration file '/WEB-INF/elasticsearch.yml' couldn't be found");
+    }
+    try {
       settings.loadFromStream("/WEB-INF/elasticsearch.yml", resourceAsStream);
-      try {
-        resourceAsStream.close();
-      } catch (IOException e) {
-        // ignore
-      }
+      resourceAsStream.close();
+    } catch (IOException e) {
+      throw new ServletException("Error while initializing elasticsearch node '" + getServletName() + "'", e);
     }
 
-    if (settings.get("http.enabled") == null) {
-      settings.put("http.enabled", false);
+    String pathData = settings.get("path.data");
+    if (pathData == null) {
+      settings.put("path.data", System.getProperty("exo.data.dir"));
+    } else if (pathData.startsWith("${")) {
+      pathData = pathData.replace("${", "");
+      pathData = pathData.replace("}", "");
+      settings.put("path.data", System.getProperty(pathData));
```

Comment:
I modified the implementation to make it more generic and flexible. (even with variables cascade `${path.data:${path.custom\}/custom}/es` )